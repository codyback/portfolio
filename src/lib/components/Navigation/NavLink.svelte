<script lang="ts">
  import { page } from '$app/stores';

  export let href: string;
  export let title: string;
  export let label: string;
</script>

<a
  {href}
  class="px-3 py-3 rounded-md hover:bg-neutral-200 dark:hover:bg-dark-700 leading-10"
  class:underline={$page.route.id?.startsWith(href)}
  {title}
  aria-label={title}
>
  {label}
</a>
