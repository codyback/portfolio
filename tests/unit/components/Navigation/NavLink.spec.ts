import NavLink from '@/lib/components/Navigation/NavLink.svelte';
import { render } from '@testing-library/svelte';
import { vi } from 'vitest';

describe('Header', () => {
	beforeAll(() => {
		vi.resetAllMocks();
	});

	it('renders correctly', () => {
		const { container } = render(NavLink, {
			href: '/test',
			title: 'Testing Navlink',
			label: 'Testing',
		});

		expect(container).toMatchSnapshot();
	});

	it('renders correctly with about page being active', () => {
		const { container } = render(NavLink, {
			href: '/about',
			title: 'About Page',
			label: 'About',
		});

		expect(container).toMatchSnapshot();
	});
});
