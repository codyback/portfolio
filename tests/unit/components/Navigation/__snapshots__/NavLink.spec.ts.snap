// Vitest Snapshot v1

exports[`Header > renders correctly 1`] = `
<body>
  <div>
    <a
      aria-label="Testing Navlink"
      class="px-3 py-3 rounded-md hover:bg-neutral-200 dark:hover:bg-dark-700 leading-10"
      href="/test"
      title="Testing Navlink"
    >
      Testing
    </a>
    <!--&lt;NavLink&gt;-->
  </div>
</body>
`;

exports[`Header > renders correctly with about page being active 1`] = `
<body>
  <div>
    <a
      aria-label="About Page"
      class="px-3 py-3 rounded-md hover:bg-neutral-200 dark:hover:bg-dark-700 leading-10 underline"
      href="/about"
      title="About Page"
    >
      About
    </a>
    <!--&lt;NavLink&gt;-->
  </div>
</body>
`;
